### PR TITLE
FT: Add replication commits

### PR DESCRIPTION
**Pull request by bennettbuchanan**

### CODE REVIEW

**Comment by bennettbuchanan** [comment]
@ironman-machine try

File: lib/api/apiUtils/object/replication.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * @return {undefined}
+ */
+function buildReplicationInfo(obj, objKey, bucketMD, content) {
+    const config = bucketMD.getReplicationConfiguration();
+    // If bucket does not have a replication configuration, do not replicate.
+    if (config) {
+        const rule = config.rules.find(rule => objKey.startsWith(rule.prefix));
+        if (rule) {
+            // eslint-disable-next-line no-param-reassign
+            obj.replicationInfo = {
```
Comment (LaurenSpiegel) [inline lib/api/apiUtils/object/replication.js]: ugh, side effects. Do we at least have a setter for this?

File: lib/api/apiUtils/object/replication.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                storageClass: rule.storageClass || '',
+            };
+        }
+    }
+    return undefined;
+}
+
+/**
+ * Set the object replicationInfo to replicate data and metadata, or only
+ * metadata if the object is a delete marker
```
Comment (LaurenSpiegel) [inline lib/api/apiUtils/object/replication.js]: what if it's a 0 byte object?

File: lib/api/apiUtils/object/replication.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+/**
+ * Set the object replicationInfo to replicate only metadata
+ * @param {object} obj - The object to set replicationInfo for
+ * @param {string} objKey - The key of the object
+ * @param {object} bucketMD - The bucket metadata
+ * @return {undefined}
+ */
+function buildReplicationInfoForObjectMD(obj, objKey, bucketMD) {
```
Comment (LaurenSpiegel) [inline lib/api/apiUtils/object/replication.js]: why do we need both this function and the one above?

File: lib/metadata/models/object/ObjectMD.js [hunk 4]
```diff
@@ -571,6 +577,32 @@ module.exports = class ObjectMD {
     }
 
     /**
+     * Set replication information
+     *
+     * @param {object} replicationInfo - replication information object
+     * @return {ObjectMD} itself
+     */
+    setReplicationInfo(replicationInfo) {
```
Comment (LaurenSpiegel) [inline lib/metadata/models/object/ObjectMD.js]: why not using this above?

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
E2E passed.

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Pull request merged**